### DISCUSSION TITLE
Guide: security-openid-connect-client,  Failed to start application (with profile dev): io.quarkus.oidc.common.runtime.OidcEndpointAccessException

### DISCUSSION THREAD

**Discussion by mike-hampton-float** [quarkusio/quarkus#27116] [Q&A] [answered]
Hi I am following https://quarkus.io/guides/security-openid-connect-client#configuring-the-application, and tried to start the example.  I got to the point of starting the server (https://quarkus.io/guides/security-openid-connect-client#keycloak-dev-mode), but it fails with 
```
2022-08-03 14:51:39,563 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): io.quarkus.oidc.common.runtime.OidcEndpointAccessException
```
I made sure to copy the example `quarkus-realm.json` into the classpath (`target/classes/quarkus-realm.json`).

I have made no modifications to the example, and even compared my repo to the solution.

Any pointers greatly appreciated.

## Edit 1

Keycloak did start, in this case at http://localhost:59830

Attempting a restart of quarkus results in :

```
[Exception 1] io.quarkus.oidc.client.OidcClientException: OIDC server is not available at the 'http://localhost:59830/realms/quarkus' URL. Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example: 'https://localhost:8180/auth/realms/quarkus'
```

I tried to manually import the `quarkus-realm.json` to the running keycloak at http://localhost:59830, and it failed to import.  

```
Error! realm role 'admin' already exists.
```

I tried to import with overwrite, and it failed again:

```
Error! App doesn't exist in role definitions: realm-management
```

[a comment by an automated account is not shown]

**Comment by mike-hampton-float** [comment, marked as answer]
I created the realm named 'quarkus' in the keycloak console.  I then imported `quarkus-realm.json` using "If a resource exists " = Skip.  I then forced a restart of quarkus.

**Reply by sberyozkin** [reply]
I think I've forgotten to sync the recent updates to the application.properties we did in the actual quickstart with the doc, will fix once I get back from  PTO, have a look at those properties please, there are some updates there to make sure the port is always correct,

**Comment by sberyozkin** [comment]
@mike-hampton-float I've just tried this quickstart, against the latest main, it works exactly as described in https://quarkus.io/guides/security-openid-connect-client#keycloak-dev-mode for me
